### ISSUE TITLE
BestCkptSaverHook 不起作用

### ISSUE THREAD

**Issue by wcf653422590** [modelscope/modelscope#699]
Thanks for your error report and we appreciate it a lot.

**Checklist**

* I have searched the tutorial on modelscope  [doc-site](https://modelscope.cn/docs)
* I have searched related issues but cannot get the expected help.
* The bug has not been fixed in the latest version.

**Describe the bug**
我利用json文件进行参数配置，在"train"模块配置了"hooks"，"hooks"中配置了BestCkptSaverHook，json文件具体是这样的：
{
    "framework": "pytorch",
    "task": "text-ranking",
    "preprocessor": {
        "type": "text-ranking"
    },
    "dataset":{
        "train": {
            "type": "bert",                             
            "query_sequence": "query",                  
            "pos_sequence": "positive_passages",        
            "neg_sequence": "negative_passages",        
            "text_fileds": ["text"],    
            "qid_field": "query_id"                     
        },
        "val": {
            "type": "bert",
            "query_sequence": "query",
            "pos_sequence": "positive_passages",
            "neg_sequence": "negative_passages",
            "text_fileds": ["text"],
            "qid_field": "query_id"
        }
    },
    "model": {
        "architectures": [
            "BertForSequenceClassification"
        ],
        "type": "bert",            
        "attention_probs_dropout_prob": 0.1, 
        "classifier_dropout": 0.0,           
        "hidden_act": "gelu",                
        "hidden_size": 768,                  
        "initializer_range": 0.02,           
        "intermediate_size": 3072,           
        "max_position_embeddings": 512,      
        "num_attention_heads": 12,           
        "num_hidden_layers": 12,             
        "pad_token_id": 0,                   
        "position_embedding_type": "absolute", 
        "vocab_size": 30522                    
    },
    "train": {
        "work_dir": "work_dir/pos2_neg0_mrr10", 
        "max_epochs": 2,   
        "neg_samples": 4,
        "train_batch_size": 4,
        "dataloader": {
            "batch_size_per_gpu": 4,
            "workers_per_gpu": 0
        },
        "optimizer": {
            "type": "AdamW",
            "lr": 1.0e-3
        },
        "lr_scheduler": {
            "type": "StepLR",
            "step_size": 2,
            "options": {
                "warmup": {
                    "type": "LinearWarmup",
                    "warmup_iters": 2
                }
            }
        },
        "hooks": [{
            "type": "TextLoggerHook",
            "interval": 1
        }, {
            "type": "IterTimerHook"
        }, {
            "type": "EvaluationHook",
            "by_epoch": false,
            "interval": 241,
            "shuffle": false
        }, {
            "type": "LrSchedulerHook",
            "warmup": {
                "type": "LinearWarmup",
                "warmup_iters": 2
            }
        },{
            "type": "BestCkptSaverHook",
            "rule": "max",
            "metric_key": "mrr@10",
            "by_epoch": false,
            "interval": 100
        }]
    },
    
    "evaluation": {
        "metrics": ["mrr@10"],
        "dataloader": {
            "batch_size_per_gpu":30,
            "workers_per_gpu": 0
        }
    }
}

**To Reproduce**
* What command or script did you run?
> 
import os
os.environ['CUDA_VISIBLE_DEVICES'] = "3"                                                                                                                                                          
# 需在GPU环境运行
from sklearn.metrics import ndcg_score
from modelscope.metainfo import Trainers
from modelscope.msdatasets import MsDataset
from modelscope.trainers import build_trainer
import tempfile
from torch.optim.lr_scheduler import StepLR
from torch.optim import AdamW
from modelscope.models import Model
import json
# load dataset
config_name = 'config/correlation.json'

train_ds = MsDataset.load('data/train.jsonl').to_hf_dataset()
dev_ds = MsDataset.load('data/dev.jsonl').to_hf_dataset()
model_id = 'damo/nlp_corom_passage-ranking_chinese-base-ecom'

kwargs = dict(
    model=model_id,
    train_dataset=train_ds,
    eval_dataset=dev_ds,
    device="gpu",
    cfg_file=config_name
    )
trainer = build_trainer(name=Trainers.nlp_text_ranking_trainer, default_args=kwargs)
trainer.train()
* Did you make any modifications on the code or config? Did you understand what you have modified?
使用的是https://www.modelscope.cn/models/damo/nlp_corom_passage-ranking_chinese-base-ecom/summary的模型训练示例代码，其中的配置相关，换成了上述的json文件
* What dataset did you use?
私人的

**Your Environments (__required__)**
* OS: linux
* CPU: use gpu

* You may add addition that may be helpful for locating the problem, such as
    * modelscope 1.10

Please @ corresponding people according to your problem:
Finetune related: @tastelikefeet  @Jintao-Huang

**Comment by slin000111** [comment]
加一下CheckpointHook，EvaluationHook、BestCkptSaverHook、CheckpointHook interval保持一致。

**Comment by wcf653422590** [comment]
谢谢回复！
经过您的提示，我做了更改，但是仍然没有效果，目前output文件夹下生成了pytorch_model.bin文件，但是output_best文件夹下却没有生成该文件。我目前的json文件更改成这样：
{
    "framework": "pytorch",
    "task": "text-ranking",
    "preprocessor": {
        "type": "text-ranking"
    },
    "dataset":{
        "train": {
            "type": "bert",                             
            "query_sequence": "query",                  
            "pos_sequence": "positive_passages",        
            "neg_sequence": "negative_passages",        
            "text_fileds": ["text"],    
            "qid_field": "query_id"                     
        },
        "val": {
            "type": "bert",
            "query_sequence": "query",
            "pos_sequence": "positive_passages",
            "neg_sequence": "negative_passages",
            "text_fileds": ["text"],
            "qid_field": "query_id"
        }
    },
    "model": {
        "architectures": [
            "BertForSequenceClassification"
        ],
        "type": "bert",            
        "attention_probs_dropout_prob": 0.1, 
        "classifier_dropout": 0.0,           
        "hidden_act": "gelu",                
        "hidden_size": 768,                  
        "initializer_range": 0.02,           
        "intermediate_size": 3072,           
        "max_position_embeddings": 512,      
        "num_attention_heads": 12,           
        "num_hidden_layers": 12,             
        "pad_token_id": 0,                   
        "position_embedding_type": "absolute", 
        "vocab_size": 30522                    
    },
    "train": {
        "work_dir": "work_dir/pos2_neg01", 
        "max_epochs": 1,   
        "neg_samples": 4,
        "train_batch_size": 4,
        "dataloader": {
            "batch_size_per_gpu": 4,
            "workers_per_gpu": 0
        },
        "optimizer": {
            "type": "AdamW",
            "lr": 1.0e-3
        },
        "lr_scheduler": {
            "type": "StepLR",
            "step_size": 2,
            "options": {
                "warmup": {
                    "type": "LinearWarmup",
                    "warmup_iters": 2
                }
            }
        },
        "hooks": [{
            "type": "TextLoggerHook",
            "interval": 1
        }, {
            "type": "IterTimerHook"
        }, {
            "type": "EvaluationHook",
            "by_epoch": false,
            "interval": 43
        }, {
            "type": "LrSchedulerHook",
            "warmup": {
                "type": "LinearWarmup",
                "warmup_iters": 2
            }
        },{
            "type": "CheckpointHook",
            "by_epoch": false,
            "interval": 43
        },{
            "type": "BestCkptSaverHook",
            "rule": "max",
            "metric_key": "mrr@10",
            "by_epoch": false,
            "interval": 43
        }]
    },
    
    "evaluation": {
        "metrics": ["mrr@10"],
        "dataloader": {
            "batch_size_per_gpu":30,
            "workers_per_gpu": 0,
            "shuffle": false
        }
    }
}
